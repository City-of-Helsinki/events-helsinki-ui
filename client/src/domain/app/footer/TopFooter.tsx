import React, { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router";

import { ReactComponent as CultureIcon } from "../../../assets/icons/svg/culture.svg";
import { ReactComponent as DanceIcon } from "../../../assets/icons/svg/dance.svg";
import { ReactComponent as FoodIcon } from "../../../assets/icons/svg/food.svg";
import { ReactComponent as InfluenceIcon } from "../../../assets/icons/svg/influence.svg";
import { ReactComponent as MovieIcon } from "../../../assets/icons/svg/movie.svg";
import { ReactComponent as MuseumIcon } from "../../../assets/icons/svg/museum.svg";
import { ReactComponent as MusicIcon } from "../../../assets/icons/svg/music.svg";
import { ReactComponent as NatureIcon } from "../../../assets/icons/svg/nature.svg";
import { ReactComponent as SearchIcon } from "../../../assets/icons/svg/search.svg";
import { ReactComponent as SmileIcon } from "../../../assets/icons/svg/smile.svg";
import { ReactComponent as SportIcon } from "../../../assets/icons/svg/sport.svg";
import { ReactComponent as TheatreIcon } from "../../../assets/icons/svg/theatre.svg";
import CategoryFilters from "../../../common/components/category/CategoryFilters";
import IconLink from "../../../common/components/link/IconLink";
import { Category } from "../../../common/types";
import { CATEGORIES } from "../../../constants";
import { getSearchQuery } from "../../../util/searchUtils";
import Container from "../layout/Container";
import styles from "./topFooter.module.scss";

const TopFooter: FunctionComponent = () => {
  const { t } = useTranslation();
  const { push } = useHistory();

  const handleCategoryClick = (category: Category) => {
    const search = getSearchQuery({
      categories: [category],
      dateTypes: [],
      endDate: null,
      isCustomDate: false,
      search: "",
      startDate: null
    });

    push({ pathname: "search", search });
  };

  return (
    <footer className={styles.topFooterWrapper}>
      <Container>
        <div className={styles.companyInfoWrapper}>
          <div className={styles.logoWrapper}>
            <div className={styles.helsinkiLogo}></div>
            <h2 className={styles.appName}>{t("appName")}</h2>
          </div>
          <div className={styles.iconLinkWrapper}>
            <IconLink
              icon={<SearchIcon />}
              text={t("footer.searchEvents")}
              to="/search"
            />
            <IconLink
              icon={<SmileIcon />}
              text={t("footer.searchCollections")}
              to="/search"
            />
          </div>
        </div>
        <div className={styles.categoriesWrapper}>
          <h2 className={styles.categoriesTitle}>
            {t("footer.titleCategories")}
          </h2>

          <CategoryFilters
            categories={[
              {
                icon: <MovieIcon />,
                text: t("home.category.movie"),
                value: CATEGORIES.MOVIE
              },
              {
                icon: <MusicIcon />,
                text: t("home.category.music"),
                value: CATEGORIES.MUSIC
              },
              {
                icon: <SportIcon />,
                text: t("home.category.sport"),
                value: CATEGORIES.SPORT
              },
              {
                icon: <MuseumIcon />,
                text: t("home.category.museum"),
                value: CATEGORIES.MUSEUM
              },
              {
                icon: <DanceIcon />,
                text: t("home.category.dance"),
                value: CATEGORIES.DANCE
              },
              {
                icon: <CultureIcon />,
                text: t("home.category.culture"),
                value: CATEGORIES.CULTURE
              },
              {
                icon: <NatureIcon />,
                text: t("home.category.nature"),
                value: CATEGORIES.NATURE
              },
              {
                icon: <InfluenceIcon />,
                text: t("home.category.influence"),
                value: CATEGORIES.INFLUENCE
              },
              {
                icon: <TheatreIcon />,
                text: t("home.category.theatre"),
                value: CATEGORIES.THEATRE
              },
              {
                icon: <FoodIcon />,
                text: t("home.category.food"),
                value: CATEGORIES.FOOD
              }
            ]}
            hasHorizontalPadding={true}
            onClickCategory={handleCategoryClick}
          />
        </div>
      </Container>
    </footer>
  );
};

export default TopFooter;
