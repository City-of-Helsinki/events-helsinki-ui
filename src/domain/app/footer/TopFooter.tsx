import { IconFill, IconFood, IconSmile } from "hds-react";
import React, { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router";

import { ReactComponent as CultureIcon } from "../../../assets/icons/svg/culture.svg";
import { ReactComponent as DanceIcon } from "../../../assets/icons/svg/dance.svg";
import { ReactComponent as MovieIcon } from "../../../assets/icons/svg/movie.svg";
import { ReactComponent as MuseumIcon } from "../../../assets/icons/svg/museum.svg";
import { ReactComponent as MusicIcon } from "../../../assets/icons/svg/music.svg";
import { ReactComponent as NatureIcon } from "../../../assets/icons/svg/nature.svg";
import { ReactComponent as SearchIcon } from "../../../assets/icons/svg/search.svg";
import { ReactComponent as SportIcon } from "../../../assets/icons/svg/sport.svg";
import { ReactComponent as TheatreIcon } from "../../../assets/icons/svg/theatre.svg";
import CategoryFilters from "../../../common/components/category/CategoryFilters";
import IconLink from "../../../common/components/link/IconLink";
import { Category } from "../../../common/types";
import { CATEGORIES } from "../../../constants";
import getLocale from "../../../util/getLocale";
import { getSearchQuery } from "../../../util/searchUtils";
import Container from "../layout/Container";
import styles from "./topFooter.module.scss";

const TopFooter: FunctionComponent = () => {
  const { t } = useTranslation();
  const locale = getLocale();
  const { push } = useHistory();

  const handleCategoryClick = (category: Category) => {
    const search = getSearchQuery({
      categories: [category.value],
      dateTypes: [],
      districts: [],
      endDate: null,
      isCustomDate: false,
      keywords: [],
      places: [],
      publisher: null,
      search: "",
      startDate: null
    });

    push({ pathname: `/${locale}/events`, search });
  };

  return (
    <footer className={styles.topFooterWrapper}>
      <Container>
        <div className={styles.companyInfoWrapper}>
          <div className={styles.logoWrapper}>
            <div className={styles.helsinkiLogo}></div>
            <h2 className={styles.appName}>{t("appName")}</h2>
          </div>
          <div className={styles.iconLinkWrapper}>
            <IconLink
              icon={<SearchIcon />}
              text={t("footer.searchEvents")}
              to={`/${locale}/events`}
            />
            <IconLink
              icon={<IconSmile />}
              text={t("footer.searchCollections")}
              to={`/${locale}/collections`}
            />
          </div>
        </div>
        <div className={styles.categoriesWrapper}>
          <h2 className={styles.categoriesTitle}>
            {t("footer.titleCategories")}
          </h2>

          <CategoryFilters
            categories={[
              {
                icon: <MovieIcon />,
                text: t("home.category.movie"),
                value: CATEGORIES.MOVIE
              },
              {
                icon: <MusicIcon />,
                text: t("home.category.music"),
                value: CATEGORIES.MUSIC
              },
              {
                icon: <SportIcon />,
                text: t("home.category.sport"),
                value: CATEGORIES.SPORT
              },
              {
                icon: <MuseumIcon />,
                text: t("home.category.museum"),
                value: CATEGORIES.MUSEUM
              },
              {
                icon: <DanceIcon />,
                text: t("home.category.dance"),
                value: CATEGORIES.DANCE
              },
              {
                icon: <CultureIcon />,
                text: t("home.category.culture"),
                value: CATEGORIES.CULTURE
              },
              {
                icon: <NatureIcon />,
                text: t("home.category.nature"),
                value: CATEGORIES.NATURE
              },
              {
                icon: <IconFill />,
                text: t("home.category.influence"),
                value: CATEGORIES.INFLUENCE
              },
              {
                icon: <TheatreIcon />,
                text: t("home.category.theatre"),
                value: CATEGORIES.THEATRE
              },
              {
                icon: <IconFood />,
                text: t("home.category.food"),
                value: CATEGORIES.FOOD
              }
            ]}
            hasHorizontalPadding={true}
            onClickCategory={handleCategoryClick}
          />
        </div>
      </Container>
    </footer>
  );
};

export default TopFooter;
