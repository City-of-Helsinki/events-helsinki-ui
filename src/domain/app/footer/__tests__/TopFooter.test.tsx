import * as React from "react";
import { MemoryRouter } from "react-router";
import renderer from "react-test-renderer";

import TopFooter from "../TopFooter";

test("TopFooter matches snapshot", () => {
  const component = renderer.create(
    <MemoryRouter>
      <TopFooter />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
