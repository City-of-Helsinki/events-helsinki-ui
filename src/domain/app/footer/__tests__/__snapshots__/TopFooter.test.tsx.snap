// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopFooter matches snapshot 1`] = `
<footer
  className="topFooterWrapper"
>
  <div
    className="container"
  >
    <div
      className="companyInfoWrapper"
    >
      <div
        className="logoWrapper"
      >
        <div
          className="helsinkiLogo"
        />
        <h2
          className="appName"
        >
          Tapahtumat
        </h2>
      </div>
      <div
        className="iconLinkWrapper"
      >
        <div
          className="iconLink"
        >
          <a
            href="/fi/events"
            onClick={[Function]}
          >
            <div
              className="iconWrapper"
            >
              <svg>
                search.svg
              </svg>
            </div>
            <div
              className="textWrapper"
            >
              Etsi tekemist√§
            </div>
          </a>
        </div>
        <div
          className="iconLink"
        >
          <a
            href="/fi/collections"
            onClick={[Function]}
          >
            <div
              className="iconWrapper"
            >
              <svg
                className=""
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.013717,1.01899281 C18.0697842,1.02004796 22.9794245,5.92968825 22.9794245,11.9852278 C22.9794245,18.0407674 18.0682014,22.9504077 12.013717,22.9504077 C5.95923261,22.9504077 1.04853717,18.041295 1.04853717,11.9852278 C1.04853717,5.92916067 5.95923261,1.02004796 12.013717,1.01899281 L12.013717,1.01899281 Z M12.013717,2.84652278 C6.97534772,2.84652278 2.87764988,6.94685851 2.87764988,11.9852278 C2.87764988,17.0235971 6.97640288,21.1228777 12.0152998,21.1228777 C17.0541966,21.1228777 21.1534772,17.0235971 21.1534772,11.9852278 C21.1534772,6.94685851 17.0526139,2.84652278 12.013717,2.84652278 Z M16.3688729,14.1029257 L18.0117506,14.903789 C16.9025481,17.205879 14.5690751,18.6656081 12.013717,18.6559233 C9.45683429,18.6671944 7.12141723,17.2069429 6.01199041,14.9032614 L7.65486811,14.1029257 C8.46605189,15.7702798 10.157662,16.828478 12.0118705,16.828478 C13.866079,16.828478 15.5576891,15.7702798 16.3688729,14.1029257 Z M8.0542446,8.33016787 C8.89543918,8.33016787 9.57736211,9.0120908 9.57736211,9.85328537 C9.57736211,10.6944799 8.89543918,11.3764029 8.0542446,11.3764029 C7.21305003,11.3764029 6.5311271,10.6944799 6.5311271,9.85328537 C6.5311271,9.0120908 7.21305003,8.33016787 8.0542446,8.33016787 Z M15.973717,8.33016787 C16.8149116,8.33016787 17.4968345,9.0120908 17.4968345,9.85328537 C17.4968345,10.6944799 16.8149116,11.3764029 15.973717,11.3764029 C15.1325225,11.3764029 14.4505995,10.6944799 14.4505995,9.85328537 C14.4505995,9.0120908 15.1325225,8.33016787 15.973717,8.33016787 Z"
                />
              </svg>
            </div>
            <div
              className="textWrapper"
            >
              Selaa kokoelmia
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="categoriesWrapper"
    >
      <h2
        className="categoriesTitle"
      >
        Kategoriat
      </h2>
      <div
        className="categoryFilters"
      >
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              movie.svg
            </svg>
            <span>
              Elokuva
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              music.svg
            </svg>
            <span>
              Musiikki
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              sport.svg
            </svg>
            <span>
              Liikunta ja urheilu
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              museum.svg
            </svg>
            <span>
              Museot
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              dance.svg
            </svg>
            <span>
              Tanssi
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              culture.svg
            </svg>
            <span>
              Kulttuuri ja taide
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              nature.svg
            </svg>
            <span>
              Luonto ja ulkoilu
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              influence.svg
            </svg>
            <span>
              Osallistu ja vaikuta
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              theatre.svg
            </svg>
            <span>
              Teatteri
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter withHorizontalPadding"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              food.svg
            </svg>
            <span>
              Ruoka
            </span>
            <svg
              className=""
              height="100%"
              viewBox="0 0 199.404 199.404"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
